bundler/ruby_version_resolver.go

```diff
@@ -0,0 +1,58 @@
+package bundler
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+// RubyVersionResolver identifies and compares versions of Ruby used in the
+// build environment.
+type RubyVersionResolver struct {
+}
+
+// NewRubyVersionResolver initializes an instance of RubyVersionResolver.
+func NewRubyVersionResolver() RubyVersionResolver {
+	return RubyVersionResolver{}
+}
+
+// Lookup returns the version of Ruby installed in the build environment.
+func (r RubyVersionResolver) Lookup(workingDir string) (string, error) {
+	getRubyVersionCmd := strings.Join([]string{
+		"ruby",
+		"--version",
+	}, " ")
+	cmdStdOut, err := RunBashCmd(getRubyVersionCmd, workingDir)
+	if err != nil {
+		return "", fmt.Errorf("failed to obtain ruby version: %w: %s", err, cmdStdOut)
+	}
+
+	versions := regexp.MustCompile(`ruby (\d+\.\d+\.\d+)`).FindStringSubmatch(cmdStdOut)
```

Comment:
this RegEx matches versions like `1.9.3` but not versions like `1.9.3-p551` or versions like `jruby-9.2.17.0` which we should also support

Reply:
RVM is now a source for installed ruby version. Next regexpes are now used for detecting a ruby version:
		`(jruby-\d+\.\d+\.\d+).\d+`,
		`(jruby-\d+\.\d+)\.\d+`,
		`(jruby-head)`,
		`(ruby-\d+\.\d+)\.\d+`,
		`(ruby-head)`
So it covers the RVM's versions format for jruby and ruby: jruby-1.2.3.4, jruby-1.2.3, ruby-1.2.3 and `head` versions. Also all possible suffixes (aka patchlevel and prerelease) are supported but ignored.